### PR TITLE
[Data] Update data ingest release tests

### PR DESCRIPTION
**Pull request by bveeramani**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/nightly_tests/dataset/read_and_consume_benchmark.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def consume_fn(ds):
     elif args.iter_torch_batches:
 
         def consume_fn(ds):
-            for _ in ds.iter_torch_batches():
+            for _ in ds.iter_torch_batches(device="cuda"):
```
Comment (bveeramani) [inline release/nightly_tests/dataset/read_and_consume_benchmark.py:81]: Move the testing of `iter_torch_batches` with CUDA from `streaming_data_ingest_benchmark_100gb_gpu` benchmark to the `iter_torch_batches` benchmark

**Review by omatthew98 — APPROVED** [review]
Lgtm!

File: release/nightly_tests/dataset/read_and_consume_benchmark.py (hunk 1 above)
Comment (srinathk10) [inline release/nightly_tests/dataset/read_and_consume_benchmark.py:81]: Would be nice to have a block comment on what this release does does.

File: release/nightly_tests/dataset/streaming_split_benchmark.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import argparse
+from typing import Optional
+
+from benchmark import Benchmark
+import ray
+
+
```
Comment (srinathk10) [inline release/nightly_tests/dataset/streaming_split_benchmark.py:7]: Block comment on the purpose of the test will be good.

File: release/release_data_tests.yaml [hunk 3]
```diff
@@ -148,92 +148,31 @@
...
-      env: gce
-      frequency: manual
-      cluster:
-        cluster_compute: data_ingest_benchmark_compute_gpu_gce.yaml
+    - __suffix__: regular
+    - __suffix__: early_stop
+      # This test case will early stop the data ingestion iteration on the GPU actors.
+      # This is a common usage in PyTorch Lightning
+      # (https://lightning.ai/docs/pytorch/stable/common/trainer.html#limit-train-batches).
+      # There was a bug in Ray Data that caused GPU memory leak (see #3.919).
```
Comment (srinathk10) [inline release/release_data_tests.yaml:172]: Could not find 3.919

Comment (bveeramani) [reply release/release_data_tests.yaml:172]: Good catch. Looks like https://github.com/ray-project/ray/pull/42102 accidently modified it: 
```diff
- # There was a bug in Ray Data that caused GPU memoy leak (see #34819).
+ # There was a bug in Ray Data that caused GPU memoy leak (see #3.919).
```

File: release/nightly_tests/dataset/streaming_split_benchmark.py (hunk 2 above)
Comment (bveeramani) [reply release/nightly_tests/dataset/streaming_split_benchmark.py:7]: Addressed

File: release/nightly_tests/dataset/fixed_size_gpu_compute.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+cloud_id: {{env["ANYSCALE_CLOUD_ID"]}}
+region: us-west-2
+
+advanced_configurations_json:
+    IamInstanceProfile: {"Name": "ray-autoscaler-v1"}
+
```
[the remark this answers is not in the document]
Comment (bveeramani) [reply release/nightly_tests/dataset/fixed_size_gpu_compute.yaml:6]: Addressed

File: release/nightly_tests/dataset/autoscaling_gpu_compute.yaml [hunk 5]
```diff
@@ -8,7 +8,7 @@ advanced_configurations_json:
 
 head_node_type:
     name: head-node
-    instance_type: m5.2xlarge
+    instance_type: g4dn.2xlarge
```
[the remark this answers is not in the document]
Comment (bveeramani) [reply release/nightly_tests/dataset/autoscaling_gpu_compute.yaml:13]: Addressed

File: release/nightly_tests/dataset/read_and_consume_benchmark.py (hunk 1 above)
Comment (bveeramani) [reply release/nightly_tests/dataset/read_and_consume_benchmark.py:83]: Added a comment for this branch

**Pull request merged**